ShenkinStore/Controllers/ShoppingCartController.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using ShenkinStore.Models;
+using ShenkinStore.ViewModels;
+
+namespace ShenkinStore.Controllers
+{
+    public class ShoppingCartController : Controller
+    {
+        ShenkinContext db = new ShenkinContext();
+
+        public string shoppingCartId { get; set; }
+
+
+        // GET: ShoppingCart
+
+        public ActionResult Index()
+        {
+
+            var userID = HttpContext.Session.GetString("UserID");
+            if (userID != null)
+            {
+                var cart = ShoppingCart.GetCart(userID.ToString());
+
+                var viewModel = new ShoppingCartViewModel
+                {
+                    CartItems = cart.GetCartItems(),
+                    CartTotal = cart.GetTotal(),
+                    Amounts = cart.GetAmount(),
+                    TotalPricePerProduct = cart.GetProductTotal()
+                };
+                return View(viewModel);
+            }
+            else
+            {
+                return RedirectToAction("Index", "Home");
+            }
+        }
+
+
+        // GET: ShoppingCart/AddToCart/id
+        public ActionResult AddToCart(int id, int amount)
+        {
+            var userID = HttpContext.Session.GetString("UserID");
+            if (userID != null)
+            {
+                var addedProduct = db.Products.Single(product => product.ProductId == id);
+                var cart = ShoppingCart.GetCart(userID.ToString());
+
+
+                cart.AddToCart(addedProduct, amount);
+
+
+
+
+
+                return RedirectToAction("Index", "ShoppingCart");
+            }
+            else
+            {
+                return RedirectToAction("Login", "Users");
+            }
+        }
+
+
+        public ActionResult IncreaseProduct(int id)
```

Comment:
Change the name to something more understandable like (IncreaseProductAmount or something like that)